From 9093ca24387ea61ad811a5287eda9c07dd121a2a Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:07:32 -0400
Subject: [powerpc] eeh: Output PHB diag-data

Message-id: <1410545655-205645-224-git-send-email-dzickus@redhat.com>
Patchwork-id: 94653
O-Subject: [RHEL7 PATCH 223/626] powerpc/eeh: Output PHB diag-data
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 2c77e95741e1202f993a4126f1f401459f1bcd4d
Author: Gavin Shan <shangw@linux.vnet.ibm.com>
Date:   Fri Nov 22 16:28:46 2013 +0800

    powerpc/eeh: Output PHB diag-data

    When hitting frozen PE or fenced PHB, it's always indicative to
    have dumped PHB diag-data for further analysis and diagnosis.
    However, we never dump that for the cases. The patch intends to
    dump PHB diag-data at the backend of eeh_ops::get_log() for PowerNV
    platform.

    Signed-off-by: Gavin Shan <shangw@linux.vnet.ibm.com>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/platforms/powernv/eeh-ioda.c b/arch/powerpc/platforms/powernv/eeh-ioda.c
index be33a16..31320c6 100644
--- a/arch/powerpc/platforms/powernv/eeh-ioda.c
+++ b/arch/powerpc/platforms/powernv/eeh-ioda.c
@@ -588,11 +588,8 @@ static int ioda_eeh_get_log(struct eeh_pe *pe, int severity,
   return -EIO;
  }
 
- /*
-  * FIXME: We probably need log the error in somewhere.
-  * Lets make it up in future.
-  */
- /* pr_info("%s", phb->diag.blob); */
+ /* The PHB diag-data is always indicative */
+ pnv_pci_dump_phb_diag_data(hose, phb->diag.blob);
 
  spin_unlock_irqrestore(&phb->lock, flags);
 
-- 
1.7.1